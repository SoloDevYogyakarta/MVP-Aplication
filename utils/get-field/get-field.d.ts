import { GetFieldType } from 'src/types/get-field.type';
export declare function getField(filename?: GetFieldType): any;
