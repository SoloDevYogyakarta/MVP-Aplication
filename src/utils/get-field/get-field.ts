import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
