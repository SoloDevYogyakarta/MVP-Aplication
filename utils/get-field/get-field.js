"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getField = void 0;
const system_1 = require("../system/system");
function getField(filename) {
    try {
        const inc = ['token', 'fake'];
        if (inc.includes(filename)) {
            return (0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`);
        }
        return JSON.parse((0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`));
    }
    catch (err) {
        throw new Error(err);
    }
}
exports.getField = getField;
//# sourceMappingURL=get-field.js.map