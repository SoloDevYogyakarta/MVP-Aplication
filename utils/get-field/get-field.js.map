{"version":3,"file":"get-field.js","sourceRoot":"","sources":["../../../src/utils/get-field/get-field.ts"],"names":[],"mappings":";;;AACA,6CAA4C;AAE5C,SAAgB,QAAQ,CAAC,QAAuB;IAC9C,IAAI;QACF,MAAM,GAAG,GAAG;YACV,MAAM;YACN,OAAO;YACP,cAAc;YACd,cAAc;YACd,gBAAgB;SACjB,CAAC;QACF,IAAI,GAAG,CAAC,QAAQ,CAAC,QAAQ,CAAC,EAAE;YAC1B,OAAO,IAAA,iBAAQ,EAAC,kBAAkB,QAAQ,MAAM,CAAC,CAAC;SACnD;QACD,OAAO,IAAI,CAAC,KAAK,CAAC,IAAA,iBAAQ,EAAC,kBAAkB,QAAQ,MAAM,CAAC,CAAC,CAAC;KAC/D;IAAC,OAAO,GAAG,EAAE;QACZ,MAAM,IAAI,KAAK,CAAC,GAAG,CAAC,CAAC;KACtB;AACH,CAAC;AAhBD,4BAgBC"}