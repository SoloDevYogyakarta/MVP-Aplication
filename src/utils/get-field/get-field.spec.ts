import { getField } from './get-field';

describe('get-field', () => {
  it('should to be defined', () => expect(getField).toBeDefined());

  it('render correctly', () => expect(getField).toMatchSnapshot());

  it('should can read filename object.txt', () =>
    expect(getField('object')).toEqual({ message: 'works!' }));

  it('should can read filename fake.txt', () =>
    expect(getField('fake')).toEqual('fake'));

  it('should to be error', () => {
    try {
      getField();
    } catch (err) {
      expect(err.toString()).toContain('ENOENT:');
    }
  });
});
