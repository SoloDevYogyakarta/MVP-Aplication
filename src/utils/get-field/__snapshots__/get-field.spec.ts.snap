// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-field render correctly 1`] = `[Function]`;
